### ISSUE TITLE
[FEATURE]: More generalized solution for version hash and file path handling

### ISSUE THREAD

**Issue by Walther** [garden-io/garden#3487]
### Background / Motivation

We are relaxing the path requirements for `valueFiles` in https://github.com/garden-io/garden/pull/3445. This fixes https://github.com/garden-io/garden/issues/3367 but has a side-effect: the hash of the `valueFiles` might not be taken into account for the module version hash, if the file path is outside the module root.

Based on internal conversations, we're essentially trading issues here: allow users to specify the `valueFiles` in a bit more flexible manner to unblock them, while introducing the slight risk of possible version/cache invalidation issues.

### What should the user be able to do?

The user should be able to use `valueFiles` from other directories, and any changes should be properly reflected in the module version hash changing, resulting in a new deployment when needed.

### Why do they want to do this? What problem does it solve?

Smooths out possible edge cases that could cause confusing situations with cache invalidation.

### Suggested Implementation(s)

Because this would involve significant refactoring, this should be implemented after `0.13` has been merged and released.

### How important is this feature for you/your team?

:rose: It’s a nice to have, but nice things are nice :slightly_smiling_face:

**Comment by vvagaytsev** [comment]
Closing this as stale. Let's reopen if this becomes a practical issue for the Garden users.
